### PR TITLE
fix(nav): correctly apply `onClick` handlers, spread `Toolbar` props

### LINKED ISSUE #784: onClick doesn't seem to be working for navItems
## Bug 
When i add the onClick event listener it doesn't seem to do anything

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issue

Resolves #784

## Proposed change

- Call `onClick` handlers for `NavItem`s generated in `Nav` and `NavList`
- Pass-through props for `Toolbar` to support `onClick` handlers
- Update stories to reflect changes

### CODE REVIEW

**Comment by SimonFinney** [comment]
@clairek20 Thanks for your patience - Here's a PR that investigates the issues that you've been encountering. 

Would you mind reviewing the changeset and the deploy preview and let me know if it satisfies your requirements? Thank you!

**Review by clairek20 — APPROVED** [review]

**Comment by clairek20** [comment]
Hi @SimonFinney 

Thanks for looking at this - i approved there! Looks good to me!

**Comment by Yohanna** [comment]
Hi @SimonFinney, thanks for fixing the issue!

I'm not sure how to test the changes locally, but I can see on the deploy preview version that the Toolbar component props now have an `onClick` option which is exactly what we need.

Thank you! 👍🏼

File: src/components/Nav/Nav.stories.js [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+import { action } from '@storybook/addon-actions';
 import { storiesOf } from '@storybook/react';
+import React from 'react';
 
 import { components } from '../../../.storybook';
 import { Nav, NavItem, NavList } from '../..';
 
-storiesOf(components('Nav'), module).add(
-  'default',
+const { name } = Nav;
```
Comment (jendowns) [inline src/components/Nav/Nav.stories.js:13]: @SimonFinney Was this left over from something else, or am I missing context? I'm not sure what it's doing here aside from renaming two (out of several) occurrences of the word "Nav"?

File: src/components/Toolbar/_mocks_/index.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ const applicationsToGenerate = [
     ]),
   },
   {
+    onClick: console.log,
```
Comment (jendowns) [inline src/components/Toolbar/_mocks_/index.js:55]: Did you mean to use this over an "action" in storybook?

File: src/components/Nav/NavList/NavList.js [hunk 3]
```diff
@@ -50,11 +50,19 @@ export default class NavList extends Component {
   buildNewItemChild({ props }, index) {
     const { onItemClick, activeHref } = this.props;
 
+    const { onClick } = props;
```
Comment (jendowns) [inline src/components/Nav/NavList/NavList.js:53]: Could `onClick` have a propType definition? And also, what becomes of `onItemClick`? (Deprecated?)

**Review by jendowns — APPROVED** [review]
@SimonFinney Approving because it appears to solve the bug & I don't want to hold this up, but I still had a couple things I think should be addressed before merging.

File: src/components/Toolbar/_mocks_/index.js (hunk 2 above)
Comment (SimonFinney) [reply src/components/Toolbar/_mocks_/index.js:55]: Good idea!

File: src/components/Nav/NavList/NavList.js (hunk 3 above)
Comment (SimonFinney) [reply src/components/Nav/NavList/NavList.js:53]: This is the `onClick` from specific `NavItem`s which never ended up being called - As far as I can tell `onItemClick` is a more general handler for all `NavItem`s within a `NavList`

File: src/components/Nav/Nav.stories.js (hunk 1 above)
Comment (SimonFinney) [reply src/components/Nav/Nav.stories.js:13]: That's right! It was a minor refactor while I was in adding the `onClick` handler

**Pull request merged**